### DISCUSSION TITLE
How could I use Confusion matrix of monai.metrics in 3d segmentation task correctly

### DISCUSSION THREAD

**Discussion by Mentholatum** [Project-MONAI/MONAI#4804] [Q&A] [answered]
**Here is the code:**

    sen_acc = ConfusionMatrixMetric(include_background=True,
                                    metric_name="sensitivity",
                                    compute_sample=True,
                                    reduction=MetricReduction.MEAN,
                                    get_not_nans=False)
    post_label = AsDiscrete(to_onehot=args.classes)
    post_pred = AsDiscrete(argmax=True,
                           to_onehot=args.classes)
    with torch.no_grad():
        for idx, input_tuple in enumerate(val_generator):
            data, target = prepare_input(input_tuple=input_tuple, args=args)
            logits = model(data)
            val_labels_list = decollate_batch(target, detach=False)
            val_labels_convert = [post_label(val_label_tensor) for val_label_tensor in val_labels_list]
            val_outputs_list = decollate_batch(logits, detach=False)
            val_output_convert = [post_pred(val_pred_tensor) for val_pred_tensor in val_outputs_list]

            acc2 = sen_acc(y_pred=val_output_convert, y=val_labels_convert)
            acc_list2 = acc2.detach().cpu().numpy()
            avg_acc2 = np.mean([np.nanmean(l) for l in acc_list2])
            print(avg_acc2)

But after running this, I got the strange number: 524288.0.
logits (pred) shape is [1, 2, 32, 256, 256]
target (ground truth) shape is [1, 1, 32, 256, 256]
What exactly is the error? Your help is greatly appreciated!

**Comment by KumoLiu** [comment, marked as answer]
Hi @Mentholatum,
You can do like below:
```
sen_acc(y_pred=val_output_convert, y=val_labels_convert)
avg_acc2 = sen_acc.aggregate()
```

**Reply by Mentholatum** [reply]
Thanks! It works!

**Comment by Nic-Ma** [comment]
Hi @Mentholatum @KumoLiu ,

You can find the typical example of metrics API in:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/metrics/metric.py#L290

Thanks.

**Reply by Mentholatum** [reply]
Thanks! That helps a lot.
